### ISSUE TITLE
Error on following a redirection

### ISSUE THREAD

**Issue by haslinger** [edgurgel/httpoison#230]
Using httpoison 0.11.1 on top of hackney 1.7.1.

The request 
`HTTPoison.get("http://rockingb.de/feed/podcast/", [], [follow_redirect: false])`
returns with a 
```
{:ok,
 %HTTPoison.Response{body: "<html>\r\n<head><title>301 Moved Permanently</title></head>...\r\n",
  headers: [{"Server", "nginx"}, {"Date", "Tue, 07 Mar 2017 11:56:53 GMT"},
   {"Content-Type", "text/html"}, {"Content-Length", "178"},
   {"Connection", "keep-alive"},
   {"Location", "http://www.rockingb.de/feed/podcast/"}], status_code: 301}}
```
which is fine, while 
`HTTPoison.get("http://rockingb.de/feed/podcast/", [], [follow_redirect: true])` 
fails with
```
** (FunctionClauseError) no function clause matching in :hackney_headers_new.do_fold/3
      (hackney) /home/stefan/phoenix/pan/deps/hackney/src/hackney_headers_new.erl:147: :hackney_headers_new.do_fold([[{1, "Host", 119}], [{2, "Host", 119}], [{3, "Host", 119}], [{4, "Host", 46}], [{5, "Host", 114}], [{6, "Host", 111}], [{7, "Host", 99}], [{8, "Host", 107}], [{9, "Host", 105}], [{10, "Host", 110}], [{11, "Host", 103}], [{12, "Host", 98}], [{13, "Host", 46}], [{14, "Host", 100}], [{15, "Host", 101}]], #Function<6.29524015/3 in :hackney_headers_new.to_iolist/1>, [["User-Agent", ': ', "hackney/1.7.1", '\r\n']])
      (hackney) /home/stefan/phoenix/pan/deps/hackney/src/hackney_headers_new.erl:184: :hackney_headers_new.to_iolist/1
```

The request to the new location itself 
`HTTPoison.get("http://www.rockingb.de/feed/podcast/", [], [follow_redirect: true])` 
would be fine again.

Am I doing something wrong here?

**Comment by laduke** [comment]
Did you find a workaround @haslinger ?

**Comment by haslinger** [comment]
I am not using the feature, but parsing redirects myself in https://github.com/PanoptikumIO/pan/blob/master/lib/pan/parser/download.ex#L69

Btw. the original feature doesn't stop on redirects to itself.

**Comment by carpodaster** [comment]
This is an aspect of #240 which has been fixed in 0.11.2.

I got this error on a URL that took 6 (yes, why?!) redirects to return a 200 OK. If no `max_redirects` option is set, [hackney defaults to `5`](https://github.com/benoitc/hackney#automatically-follow-a-redirection). Redirects to self will trigger that max directs limit.

**Comment by edgurgel** [comment]
I'm closing this as it seems to be fixed. If it's still an issue please reopen! 🎉
